### PR TITLE
chore: updates around 2023-06-01

### PR DESCRIPTION
**Pull request by coryan**
Ubuntu 18.04 reached EOL, that bumps the minimum version for Ubuntu, CMake and Clang.

Add some footnotes to make it easier to follow the expected dates.

### CODE REVIEW

File: foundational-cxx-support-matrix.md [hunk 1]
```diff
@@ -30,6 +30,17 @@ dimension in question.
...
 tools, etc.) on RHEL 7 and CentOS 7.
 
+[^cmake]: We support the oldest version of CMake that ships with one of the
+supported distros. Currently that is CMake 3.13 as Debian 10 ships with this
+version.
+
+[^gcc]: Once RHEL 7 reaches EOL we will support the oldest version of GCC
+that ships with one of the supported distros.
+
+[^clang]: We support the oldest version of CLang that ships with one of the
```
Comment (fowles) [inline foundational-cxx-support-matrix.md:40]: you say `CLang` here, but in the table it is listed as `Clang`.  Not sure which is correct

**Review by fowles — APPROVED** [review]

File: foundational-cxx-support-matrix.md (hunk 1 above)
Comment (coryan) [reply foundational-cxx-support-matrix.md]: Thanks.  Fixed, it should be `Clang`.

File: foundational-cxx-support-matrix.md [hunk 2]
```diff
@@ -30,6 +30,17 @@ dimension in question.
...
 
+[^cmake]: We support the oldest version of CMake that ships with one of the
+supported distros. Currently that is CMake 3.13 as Debian 10 ships with this
+version.
+
+[^gcc]: Once RHEL 7 reaches EOL we will support the oldest version of GCC
+that ships with one of the supported distros.
+
+[^clang]: We support the oldest version of Clang that ships with one of the
+supported distros. Currently that is CLang 7.0 as Ubuntu 20.04 ships with
```
Comment (fowles) [inline foundational-cxx-support-matrix.md:41]: one more on this line

**Review by fowles — APPROVED** [review]

File: foundational-cxx-support-matrix.md (hunk 2 above)
Comment (coryan) [reply foundational-cxx-support-matrix.md]: groan... sorry about that.

**Pull request merged**